### DISCUSSION TITLE
weird loader behaviour

### DISCUSSION THREAD

**Discussion by acd02** [remix-run/remix#1177] [Q&A] [answered]
Hey there 👋 

I don't know if it's a bug or not, but it seems that sometimes, a loader is being called as the current route (a dynamic segment) is about to get updated, e.g.:

- I'm on `someRoute/users/1`
- I click on a link that redirects me to another dynamic segment of the same route -> `someRoute/users/2`
  - loader is called with the new id (2 in this case) as expected
  - but it is then also called with the previous id (1 in this case) 🤔

 Repro 👇 :
https://stackblitz.com/edit/node-huqahk ⚡

Video 🎥 :

![Kapture 2021-12-22 at 13 26 03](https://user-images.githubusercontent.com/51230231/147092787-3ecc3ab2-c2b5-4760-9c1b-088a351f1c76.gif)

**Comment by tigerabrodi** [comment]
@acd02 Hi :wave: 

I had a quick look into the repo due to my time being tight, but honestly, I'm not sure what is going wrong.

I have a feeling it may somehow be related to these utils: https://github.com/acd02/remix-loaders/blob/5f3ffe6b9a1da25ce45fa975cd812b4dbf85f9e2/app/utils/http.ts

You also seem to be doing some functional programming stuff here: https://github.com/acd02/remix-loaders/blob/5f3ffe6b9a1da25ce45fa975cd812b4dbf85f9e2/app/routes/users/%24userId.tsx#L33

Not sure exactly what it does, but it may be related :thinking:.

I know you probably had a deep look into this already, I'm just ranting my thoughts :orange_heart:

**Reply by acd02** [reply]
Hey @tigerabrodi , thanks for taking the time to have a look at my repo 👍 

I've removed all the FP stuff, as to avoid any confusion, but unfortunately the problem remains 😅

**Comment by acd02** [comment, marked as answer]
Closing this, as I opened an issue 👉 #1249
